### PR TITLE
sdlsample: Remove local _exit() declaration

### PR DESCRIPTION
**Pull request by dracc**
When #103 has been merged, a local `_exit()` declaration become extraneous.

### CODE REVIEW

File: samples/sdl/main.c [hunk 1]
```diff
@@ -18,10 +18,10 @@
 #include "string.h"
 #include <SDL.h>
 
-int _exit(int x)
+int Reboot(int x)
```
Comment (JayFoxRox) [inline samples/sdl/main.c]: nit: `reboot` (or make `quit` &rarr; `Quit`).

Comment (JayFoxRox) [inline samples/sdl/main.c]: You can also improve the function signature now: `static void Reboot()` or possibly `static void Reboot(unsigned int step)` (where `step` should be printed / have a unique number).

File: samples/sdl/main.c [hunk 2]
```diff
@@ -187,7 +187,7 @@ static void
 quit(int rc)
 {
     SDL_VideoQuit();
-    _exit(rc);
+    Reboot(rc);
```
Comment (JayFoxRox) [inline samples/sdl/main.c]: There's only 1 single instance where SDL video isn't initialized yet.
Consider merging `quit` and `Reboot` into one function - they do almost exactly the same.

You could just move `SDL_VideoQuit` into the caller for example.

File: samples/sdl/main.c (hunk 1 above)
Comment (JayFoxRox) [inline samples/sdl/main.c]: The name of this function is misleading. It's more of a `PrintErrorAndThenQuitAfter5Seconds`.

*I can't think of a good name myself.*

File: samples/sdl/main.c (hunk 2 above)
Comment (JayFoxRox) [reply samples/sdl/main.c]: I still think this argument is valid; especially as we use arbitrary return codes (1 and 5; whatever those mean).

Eitherway, this is "better than master" and it should be merged soon.

File: samples/sdl/main.c [hunk 3]
```diff
@@ -187,7 +186,7 @@ static void
 quit(int rc)
```
Comment (thrimbor) [inline samples/sdl/main.c]: nit: I don't consider `quit` to be a good name (the name sounds a bit like it's standard library functionality to me). Can't think of a better name right now, though.

Comment (thrimbor) [inline samples/sdl/main.c]: The parameter is unused now.

File: samples/sdl/main.c [hunk 4]
```diff
@@ -18,13 +18,12 @@
 #include "string.h"
 #include <SDL.h>
 
-int _exit(int x)
+static void printSDLErrorAndReboot(void)
 {
-    debugPrint("_exit called\n");
-    debugPrint( " SDL_Error: %s\n", SDL_GetError() );
+    debugPrint("Reboot called\n");
```
[the remark this answers is not in the document]
Comment (JayFoxRox) [reply samples/sdl/main.c]: Good solution is probably to move this print down, and have it say "Rebooting in 5 seconds", so the user is aware what's about to happen.

File: samples/sdl/main.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (JayFoxRox) [reply samples/sdl/main.c]: Also see existing discussion at https://github.com/XboxDev/nxdk/pull/104#discussion_r281714165

Comment (JayFoxRox) [reply samples/sdl/main.c]: See existing discussion at https://github.com/XboxDev/nxdk/pull/104#discussion_r281713666 (which used to be resolved by @dracc )

**Comment by JayFoxRox** [comment]
Code looks good to me now. I did not try running it.

**Comment by thrimbor** [comment]
Tested and merged. Thanks!

**Pull request merged**